### ISSUE TITLE
Channel-specific NormalizeOp support [feature request?]

### ISSUE THREAD

**Issue by ben-xD** [tensorflow/tflite-support#217]
Some models pre-process channels by different amounts, based on the average pixel values in separate channels of the training dataset. When using future data, we would have to preprocess the input data as they do: For example, preprocessing for vggface subtracts different values from the red, green and blue channels (some code [here](https://github.com/rcmalli/keras-vggface/blob/719ad75625fe51436b9a9479114b3d136847e11a/keras_vggface/utils.py#L57)).

I don't think this is supported by tf.image (because it generates pixel values which are not meaningful as images, only as arrays?).

**Unrelated:**
I was unsure what is meant [here](https://github.com/tensorflow/tflite-support/blob/3ce83f0cfe2c68fecf83e019f2acc354aaba471f/tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/common/ops/NormalizeOp.java#L53)
> Note: The returned {@link TensorBuffer} is always a {@link DataType#FLOAT32} tensor at present, except that the input is a {@link DataType#UINT8} tensor, {@code mean} is set to 0 and {@code stddev} is set to 1.

What does it mean: "except that the input is a UINT8 Tensor?
Do you mean "**expect** that the input is a UINT8 Tensor"?

**Comment by lu-wang-g** [comment]
Yes, I don't think tf.image support per channel normalization. Usually this is done simply by 
```
VGG_MEAN = [123.68, 116.78, 103.94]
input_img_norm = (input_img - tf.constant(VGG_MEAN)) / 255.0
```

> What does it mean: "except that the input is a UINT8 Tensor?
It means if the input tensor is UINT8, and if mean is 0 and stddev is 1, the returned TensorBuffer will not be a FLOAT32 tensor, but a UINT8 tensor, because the original input tensor is returned.

**Comment by ben-xD** [comment]
It looks like the codegen which uses support library already does this. After adding the appropriate metadata using Python, the relevant codegen is:
```java
.add(new NormalizeOp(metadata.getImageMean(), metadata.getImageStddev()));
```
where the generated `class Metadata` contains 
```java
        public float[] getImageMean() {
            return Arrays.copyOf(imageMean, imageMean.length);
        }
```

I will leave it open for you to update the comment? Thanks for your reply.

**Comment by lu-wang-g** [comment]
As you've noticed, Codegen does support per channel normalization through metadata, same as [TFLite Task library](https://www.tensorflow.org/lite/inference_with_metadata/task_library/overview) .

The comments have been updated. See [here](https://github.com/tensorflow/tflite-support/blob/42e121f32c2b641f0a31ffd2e82ed57196eacf92/tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/common/ops/NormalizeOp.java#L52-L54). Let me know if it makes sense to you.

**Comment by lu-wang-g** [comment]
Closing the issue. Feel free to reopen it if you have other questions.
